### ISSUE TITLE
Starting in fullscreen mode

### ISSUE THREAD

**Issue by promisedlandt** [kovidgoyal/kitty#856]
Hello,

is there a configuration option to start kitty in fullscreen mode? `remember_window_size  yes` starts in the fullscreen size, but the window is not in fullscreen state.

Also, `remember_window_size  yes` is not great for when you change resolutions (e.g. notebook built-in display and external display), since what is fullscreen size for one resolution isn't for another.

**Comment by promisedlandt** [comment]
Thank you very much. I thought it was only missing documentation :)

**Comment by kovidgoyal** [comment]
Note that the name of the command line option is changing to --start-as to allow for starting kitty maximized/minimized as well

**Comment by mateuszwieloch** [comment]
Is there a way to have kitty start in fullscreen mode right away when I run the application on macOS?
Context: my current workflow is terrible. I start kitty, and then run `kitty --start-as=fullscreen` to enter fullscreen mode (or use a keyboard shortcut). It would be much better if it could start in fullscreen right away.

**Comment by kovidgoyal** [comment]
https://sw.kovidgoyal.net/kitty/faq.html#how-do-i-specify-command-line-options-for-kitty-on-macos

**Comment by mateuszwieloch** [comment]
Thank you! I sent you a few bucks to support building the best terminal emulator yet. Keep up the good job!

**Comment by kovidgoyal** [comment]
Thanks :)

**Comment by Quantenzitrone** [comment]
how about renaming it to `--fullscreeen`?
`--fullscreen` is a rather common option for terminal emulators, as all these terminal emulators implement this interface:
cool-retro-term sakura roxterm foot konsole terminology terminator xfce4-terminal
which is 8/30 terminal emulators for linux i tested

**Comment by kovidgoyal** [comment]
--start-as is far more general than just --fullscreen. Having multiple options that do the same thing is both redundant and confusing for users. Doing that just because some other terminal emulators have the option is not convincing.

**Comment by Quantenzitrone** [comment]
i understand

**Comment by thisismygitrepo** [comment]
What about Linux users? how can they  automate --start-as=fullscreen launch command?

**Comment by PhilippFeO** [comment]
> What about Linux users? how can they automate --start-as=fullscreen launch command?

You can try modifying the `.desktop` file of kitty. [Here](https://askubuntu.com/questions/481517/run-script-full-screen-in-terminal) is an example for `gnome-terminal`. A workaround, if this fails, is triggering fullscreen with `man wmctrl` (it contains examples).

**Comment by LinuxIsCool** [comment]
Is there a way to set this in the config file?

**Comment by PhilippFeO** [comment]
> Is there a way to set this in the config file?

I don't know, but you can search the config file. Binding `kitty --start-as=fullscreen` to any key might also work.

**Comment by ryuheechul** [comment]
Hopefully in the future, there can be a setting like how Alacritty does (as that's is the easier knob than having to add a flag for a GUI app).
```toml
# from alacritty.toml
[window]
startup_mode = "Fullscreen"
```

But in the meantime I fixed my `kitty.desktop` with Nix (via home-manager) https://github.com/ryuheechul/dotfiles/commit/882945f14f8799c1b4bfa3aad2d7331495a6ce40.

The diff from the link above should give you an idea on how to fix the `kitty.desktop` directly even if you don't use nix and hopefully this is helpful for people who are not super familiar with modifying `.desktop` files yet.

Basically just change `Exec` like below.
```diff
# at [your/path/to/]kitty.desktop
- Exec=kitty
+ Exec=kitty --start-as=fullscreen
```

---

Update: now I see that there is [startup-sessions](https://sw.kovidgoyal.net/kitty/overview/#startup-sessions) which you can set `os_window_state fullscreen`. and this is [the commit](https://github.com/ryuheechul/dotfiles/commit/87a84522110441460018f0bda076d7fc2de7c0a0) that applies to my dotfiles and with this no need to use a flag like `--start-as`.
